bin/fillTestData/Experiences.js

```diff
@@ -0,0 +1,343 @@
+        console.error(
+          chalk.red(
+            `At least ${profileType.minimumTestableUserNumber()} users must exist in the db to test experiences. Please create more users and run again`,
+          ),
+        );
+        mongooseService.disconnect();
+        return;
+      }
+
+      const experience = await Reference.findOne();
+      if (experience && !argv.reCreate) {
+        console.log(
+          chalk.red(
+            'Experiences already exist in the db. Use --re-create option if you want to rewrite them',
+          ),
+        );
+      } else {
+        if (experience) {
+          console.log(
+            chalk.yellow(
+              'Running with --re-create option. Removing existing Experiences ...',
+            ),
+          );
+          await Reference.deleteMany();
+        }
+        console.log(
+          `Profile with many experiences (username): ${
+            users[profileType.getSeqNumWithManyExperiences()].username
+          }`,
+        );
+
+        try {
+          const experienceSharingMatrix = generateExperienceSharingMatrix(
+            users.length,
+            argv.averageNumberPerProfile,
+            argv.replyRate,
+          );
+          const nExperiencesToBeAdded = countTrueElements(
+            experienceSharingMatrix,
+          );
+          console.log(
+            chalk.green(`Saving ${nExperiencesToBeAdded} experiences ... `),
+          );
+
+          const saveExperiences = function* () {
+            for (let i = 0; i < experienceSharingMatrix.length; i++) {
+              if (i % 100 === 0) {
+                process.stdout.write(`Saving experiences for user ${i}...\n`);
+              }
+              for (let j = 0; j < experienceSharingMatrix[i].length; j++) {
+                if (experienceSharingMatrix[i][j]) {
+                  const experience = createExperience(users, i, j);
+                  yield saveExperience(experience);
+                }
+              }
+            }
+          };
+
+          const pool = new PromisePool(saveExperiences, 100);
+          await pool.start();
+        } catch (err) {
+          console.log(err);
+        }
+      }
+      console.log('');
+      mongooseService.disconnect();
+
+      function createExperience(users, to, from) {
+        const experience = new Reference();
+        experience.userTo = users[to]._id;
+        experience.userFrom = users[from]._id;
+        experience.created = experienceGenerator.created();
+        experience.feedbackPublic = experienceGenerator.feedbackPublic();
+        experience.interactions = experienceGenerator.interactions();
+        experience.recommend = experienceGenerator.recommendation();
+        experience.public = experienceGenerator.public();
+        return experience;
+      }
+
+      function saveExperience(experience) {
+        return new Promise((resolve, reject) => {
+          experience.save(err => {
```

Comment:
Would returning `experience.save()` here directly work without callbacks since I think it's a Promise already?